.github/workflows/ttrt-flatbuffer-ttnn.yml

```diff
@@ -0,0 +1,87 @@
+name: TTRT Flatbuffer TTNN Tests
+
+on:
+  workflow_dispatch:
+  workflow_call:
+  push:
+    branches: [ "tpatel/issue-104" ]
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        build: [
+          {name: "TTRT TTNN n150 runtime tests", runs-on: n150, c_compiler: clang, cpp_compiler: clang++, build_type: Release, enable_runtime: ON, enable_perf: OFF},
+          {name: "TTRT TTNN n300 runtime tests", runs-on: n300, c_compiler: clang, cpp_compiler: clang++, build_type: Release, enable_runtime: ON, enable_perf: OFF},
+        ]
+    name: ${{ matrix.build.name }}
+    runs-on: ${{ matrix.build.runs-on }}
+    steps:
+    - uses: actions/checkout@v4
+      with:
+          fetch-depth: 0 # Fetch all history and tags
+
+    - name: Install build dependencies
+      uses: ./.github/actions/install-deps
+      with:
+        os: ${{ matrix.build.runs-on }}
+
+    - name: Build and cache ttmlir-toolchain
+      uses: ./.github/actions/build-toolchain
+      with:
+        os: ${{ matrix.build.runs-on }}
+        sdk: ${{ env.SDK_VERSION }}
+
+    - name: Set reusable strings
+      id: strings
+      shell: bash
+      run: |
+        echo "build-output-dir=${{ github.workspace }}/build" >> "$GITHUB_OUTPUT"
+
+    - name: Configure CMake
+      shell: bash
+      run: |
+        source env/activate
+        cmake -G Ninja \
+        -B ${{ steps.strings.outputs.build-output-dir }} \
+        -DCMAKE_CXX_COMPILER=${{ matrix.build.cpp_compiler }} \
+        -DCMAKE_C_COMPILER=${{ matrix.build.c_compiler }} \
+        -DCMAKE_BUILD_TYPE=${{ matrix.build.build_type }} \
+        -DTTMLIR_ENABLE_RUNTIME=${{ matrix.build.enable_runtime }} \
+        -DTTMLIR_ENABLE_RUNTIME_TESTS=${{ matrix.build.enable_runtime }} \
+        -DTT_RUNTIME_ENABLE_PERF_TRACE=${{ matrix.build.enable_perf }} \
+        -S ${{ github.workspace }}
+
+    - name: Build
+      shell: bash
+      run: |
+        source env/activate
+        cmake --build ${{ steps.strings.outputs.build-output-dir }} --config ${{ matrix.build.build_type }}
+
+    - name: Build ttrt
+      shell: bash
+      run: |
+        source env/activate
+        cmake --build ${{ steps.strings.outputs.build-output-dir }} --config ${{ matrix.build.build_type }} -- ttrt
+
+    - name: Run Test
+      shell: bash
+      run: |
+        source env/activate
+        ttrt query --save-artifacts
+        export SYSTEM_DESC_PATH=${{ github.workspace }}/ttrt-artifacts/system_desc.ttsys
```

Comment:
do we want all the possible system descs? If so, it can be a simple ansible script that goes in to every machine provided in an inventory file and queries it. there's a GitHub api to get list of ci runners registered in mlir

Reply:
Yeah this is a good question, ideally we want all of them, but then use some policy for divvying them up amongst the tests, i.e. we want to distribute the work evenly between the pool.

I think it might make sense to save the system desc's with their product name or tag them in some way, `n300.ttsys`/`n150.ttsys`/`tgg.ttsys`, but we probably need to further specialize the name because of row harvesting.  It gets even more complicated if we have llm box, this is where system desc slicing could be used so we could partition the single host into 8 devices and go parallel.

Let's think through how to capture this information thoroughly because it's going to need to be rock solid to avoid CI device management hell.

Reply:
Another challenge, is that the system desc must be collected with the _current build_.  Maybe this works like:
- Build on GitHub runner, produces build artifact
- Run a super fast job on silicon runners that collect all the system descs into an artifact
- Go back to GitHub runner which runs compiler tests with collected system descs
- Go back to silicon runner which runs the flatbuffers

Not sure if that's too crazy going back and forth.  I'm just not sure how else it could work.

Reply:
fyi @vmilosevic

Reply:
So 2 things that I think we should workout:
1. Ideally llvm-lit tests are run on the compute only GitHub runners, but maybe it's not a big deal to just run this on the silicon runners.  But these need to run in order to generate flatbuffers.
2. Let's say we have a pool of 10 n300s, and we have 100 silicon tests.  We don't want to run all 100 tests on all n300s, we want to group the tests into 10 chunks and have each runner run only 10 tests.

I think we should establish some terminology (feel free to suggest other names if these seem confusing or don't make sense, but I think we need some terms to refer to because our test matrix is quite complicated):
- **Silicon Runner**: 1to1 mapping to a unique SystemDesc.  Because of row harvesting even different n300s might have unique system descs.
- **Target Family**: A category of system type, we should organize these by product name, like n150, n300, TG, TGG, LLM Box etc.  We will have a pool of n300's for example (all the same target system), but might each have unique system descs.
- **Target Capabilities**: high level attributes that describe _testable_ traits of a **Target Family**.
- **Test Capabilities**: A set of **System Capabilities** defined in the test.  Effectively describes which system it should be compiled and run against.


To run through an example of how this could work, tracing through the perspective from a single test:
> Please reference llvm-lit directives [REQUIRES and UNSUPPORTED](https://llvm.org/docs/TestingGuide.html#constraining-test-execution)
1. GitHub CI builds tt-mlir and ttrt
2. All **Silicon Runners** picks up the build artifact and generates all system descs for all runners in our pool.  We need to organize all of these somehow and bucketize them by **Target Family**.  Each **Target Family** bucket forms a pool of available test targets for Silicon tests to be compiled against.  Each **Target Family** bucket has an associated set of **Target Capabilities** which can be driven by the data in system desc itself (more on this later)
3. The system desc artifact (a zip file containing all system descs in the entire pool) comes back to the GitHub compute runner.  Here we do some kind of partitioning of the **Silicon Runners** within each **Target Family** category.  So for example, n300 is a **Target Family**, there might be 10 **Silicon Runners** of this type, ideally all of the tests that opt into being compiled against this **Target Family** get evenly distributed amongst the 10 available runners.
4. Zooming in on a particular test, it might have llvm-lit directive `REQUIRES: Galaxy HugePage`.  Here `Galaxy` and `HugePage` forms this test's **Test Capabilities** (just a set of 1).  The `Galaxy` tag itself is a **Target Capability** that is associated with TG and TGG because both **Target Family**'s export this **Target Capability**.  This would opt the test into being compiled against a single system desc that is part of this **Target Family** pool.

A note on **Target Capabilities**, I think some of these tags should be opt in and some opt out.  For example, most silicon tests, without any **Test Capabilities** listed should implicitly opt into running on most silicon runners, like by default everything should run on n300's. Whereas multi-device tests should explicitly opt in (via `REQUIRES` directive).